src/TransitionSystems/common.rs

```diff
@@ -153,4 +153,16 @@ impl<T: ComposedTransitionSystem> TransitionSystem for T {
     fn get_composition_type(&self) -> CompositionType {
         self.get_composition_type()
     }
+
+    fn get_combined_decls(&self) -> Declarations {
+        let (left, right) = self.get_children();
+        let mut clocks = HashMap::new();
+        let mut ints = HashMap::new();
+        for decl in [left.get_combined_decls(), right.get_combined_decls()] {
+            clocks.extend(decl.clocks);
+            ints.extend(decl.ints)
+        }
+
+        Declarations { ints, clocks }
+    }
```

Comment:
Couldn't this just be a default implementation on `TransitionSystem`, and then only create a new for `CompilledComponent`?

Reply:
Fixed @t-lohse